src/java/src/main/java/com/murex/tbw/storage/JsonRepository.java

```diff
@@ -33,8 +34,12 @@
 
     public JsonRepository() {
         invoiceMap = new HashMap<>();
-        sourceFile = new File(this.getClass().getClassLoader().getResource("repository.json").getFile());
+        final String path = "src/resources/repository.json";
+        sourceFile = Paths.get(path).toFile();
         try {
+            if (!sourceFile.exists()) {
+                throw new FileNotFoundException("repository.json cannot be found under the following path: " + path);
```

Comment:
Did you think of the case when the file is deleted after line 40 and before line 41 (very unlikely I admit ;-)) ?

Reply:
Isn't this part of the things to refactor? Because the previous implementation didn't check for that either

Reply:
Since this is a workshop, we assumed that the JSON file will be always there :) that's why we didn't add a try/catch.

Reply:
The C++ code does this check, though